mcs/class/corlib/System/TimeZoneInfo.Android.cs

```diff
@@ -560,39 +560,63 @@ static int ParseNumericZone (string name)
 				}
 			}
 			
-			[DllImport ("__Internal")]
-			static extern int monodroid_get_system_property (string name, ref IntPtr value);
-
-			[DllImport ("__Internal")]
-			static extern void monodroid_free (IntPtr ptr);
-			
 			static string GetDefaultTimeZoneName ()
 			{
-				IntPtr value = IntPtr.Zero;
-				int n = 0;
 				string defaultTimeZone  = Environment.GetEnvironmentVariable ("__XA_OVERRIDE_TIMEZONE_ID__");
-
 				if (!string.IsNullOrEmpty (defaultTimeZone))
 					return defaultTimeZone;
 
 				// Used by the tests
-				if (Environment.GetEnvironmentVariable ("__XA_USE_JAVA_DEFAULT_TIMEZONE_ID__") == null)
-					n = monodroid_get_system_property ("persist.sys.timezone", ref value);
-				
-				if (n > 0 && value != IntPtr.Zero) {
-					defaultTimeZone = (Marshal.PtrToStringAnsi (value) ?? String.Empty).Trim ();
-					monodroid_free (value);
+				if (Environment.GetEnvironmentVariable ("__XA_USE_JAVA_DEFAULT_TIMEZONE_ID__") == null) {
+					defaultTimeZone = GetPersistTimeZone ()?.Trim ();
 					if (!String.IsNullOrEmpty (defaultTimeZone))
 						return defaultTimeZone;
 				}
 				
-				defaultTimeZone = (AndroidPlatform.GetDefaultTimeZone () ?? String.Empty).Trim ();
+				defaultTimeZone = GetDefaultTimeZone ()?.Trim();
 				if (!String.IsNullOrEmpty (defaultTimeZone))
 					return defaultTimeZone;
 
 				return null;
 			}
 
+			static string GetPersistTimeZone ()
+			{
+				IntPtr id = IntPtr.Zero;
+				try {
+					int res = monodroid_get_system_property ("persist.sys.timezone", ref id);
```

Comment:
[`monodroid_get_system_property()`](https://github.com/xamarin/xamarin-android/blob/88ef36b85272206103e2013e4a8a3e061843d7c9/src/monodroid/jni/monodroid-glue.c#L298-L325) *allocates memory*, held by `id`. In particular see: https://github.com/xamarin/xamarin-android/blob/88ef36b85272206103e2013e4a8a3e061843d7c9/src/monodroid/jni/monodroid-glue.c#L317-L322

As such, if `id != IntPtr.Zero`, it *must* be free'd in order to avoid a memory leak...